File: programmers/python/입문100/9.4_구슬을나누는경우의수★/9-4.py
# 펙토리얼(재귀)
def recur(n): # n * (n-1) * .. * 2 * 1
    if n == 0 or n == 1:
        return 1
    return n * recur(n-1)

def solution(n, r):
    a = recur(n)
    b = recur(n-r)
    c = recur(r)
    return int(a / (b * c))

File: programmers/java/기초124/9.4_접미사배열/9_4.java
import java.util.Arrays;

class Solution {
    public String[] solution(String my_string) {
        // String[] answer = {};
        // return answer;
        //////////////////////////////////////
        int n = my_string.length();
        String[] answer = new String[n];

        for (int i = 0; i < n; i++) {
            answer[i] = my_string.substring(i);
        }

        Arrays.sort(answer);
        //////////////////////////////////////
        return answer;
    }
}